client/webserver/site/src/js/settings.js

```diff
@@ -109,7 +109,11 @@ export default class SettingsPage extends BasePage {
     const page = this.page
     page.exportAccountHost.textContent = host
     page.exportAccountErr.textContent = ''
-    this.showForm(authorizeAccountExportForm)
+    if (authorizeAccountExportForm.hidden) {
```

Comment:
If this is the reason you're setting the `hidden` attribute on hidden form elements, you can use `Doc.isHidden(authorizeAccountExportForm.parentElement)` instead. 

Or even better, write a function/method to tell whether there is a cached password. Perhaps a method on `State` that uses `State.getCookie` under the hood.

Reply:
The reason I like adding the hidden attribute is that there is not only a check to make sure that the cookie contains the key, but there is also a check (in `authMiddleware`) to make sure that the key can unlock a password.

Reply:
I ended up going with a new method on `State`. This is cleaner, and there shouldn't be an invalid key in the cookies unless there is a bug elsewhere.